### PR TITLE
feat: add DepositFee component in FeeReward tab

### PR DESCRIPTION
**Pull request by ChristanDaniel**
## Why

Previously, users were unable to modify fee values after creating a pool. Now, we have added a feature on the FeeRewards tab within the Manager page, allowing them to change the fee.

## Changes

- Create DepositFee component
- create setJoinFees function in useManagePoolController hook
- add DepositFee component in Create Pool 
- add DepositFee component in FeeRewards Tab

### CODE REVIEW

File: src/components/DepositFee/index.tsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  feeRate?: string
+  brokerCommision?: number
+  managerShare?: number
+}
+
+interface IDepositFeeProps {
+  feesData?: Record<string, FeesData>
+  disabledNoEvent?: boolean
+  handleClickToggle: (event: React.ChangeEvent<HTMLInputElement>) => void
+  handleRefferalCommission: (event: React.ChangeEvent<HTMLInputElement>) => void
```
Comment (nicholascostadev) [inline src/components/DepositFee/index.tsx:24]: It is spelled "referral"

```suggestion
  handleReferralCommission: (event: React.ChangeEvent<HTMLInputElement>) => void
```

File: src/components/DepositFee/index.tsx
Comment (nicholascostadev) [inline src/components/DepositFee/index.tsx:1]: You don't need to change it, but I tend to prefer having the actual event on the end of the function name, for example, you have this function

```ts
function handleClickToggle() {...}
```
Here you have this sequence on the naming:
1. Prefix -> handle
2. What(Event) -> Click
3. Where -> Toggle

I guess it's easier to understand where and what when they're on this sequence

```ts
function handleToggleClick() {...}
```
1. Prefix -> handle
2. Where -> Toggle
3. What(Event) -> Click

But It does not matter effectively, it's just my opinion

File: src/components/DepositFee/index.tsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+type FeesData = {
+  isChecked: boolean
+  feeRate?: string
+  brokerCommision?: number
+  managerShare?: number
+}
+
+interface IDepositFeeProps {
+  feesData?: Record<string, FeesData>
+  disabledNoEvent?: boolean
```
Comment (nicholascostadev) [inline src/components/DepositFee/index.tsx:22]: What does this variable do? Is it the best name you can give it?

I'm saying that outside the context of what you're building, without reading through the code, I don't get what that is supposed to do.

After looking at the Button component, I also saw a problem:
image

You have a prop `disabledNoEvent` that is doing the same purpose of the `disabled` prop, you should probably remove this `disabledNoEvent` prop and use only `disabled`, it will also be more readable.

File: src/components/DepositFee/styles.ts [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+      }
+      ${Input}[type="number"]:not([value='']) ~ ${Error} {
+        display: block;
+
+        color: ${theme.colors.amber};
+      }
+    `}
+`
+
+export const RefferalCommissionWrapper = styled.div`
```
Comment (nicholascostadev) [inline src/components/DepositFee/styles.ts:156]: It is spelled "referral"

```suggestion
export const ReferralCommissionWrapper = styled.div`
```

File: src/components/DepositFee/styles.ts [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+      transition-property: max-height opacity;
+    `}
+`
+
+interface IWrapperInputFeeProps {
+  isAddress: boolean
+  value: number
+}
+
+export const WrapperInputFee = styled.span<IWrapperInputFeeProps>`
```
Comment (nicholascostadev) [inline src/components/DepositFee/styles.ts:116]: "Wrapper" should be at the end

```suggestion
export const InputFeeWrapper = styled.span<IInputFeeWrapperProps>`
```

File: src/components/DepositFee/styles.ts [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+      color: #c4c4c4;
+      font-weight: ${theme.font.weight.normal};
+      font-size: ${theme.font.sizes.font12};
+      line-height: 1.4rem;
+      text-transform: uppercase;
+    }
+  `}
+`
+
+export const WrapperInputRange = styled.div`
```
Comment (nicholascostadev) [inline src/components/DepositFee/styles.ts:204]: "Wrapper" should be at the end

```suggestion
export const InputRangeWrapper = styled.div`
```

File: src/templates/Manage/CreatePool/ConfigureFee/FeeConfig/index.tsx [hunk 6]
```diff
@@ -47,7 +47,7 @@ const FeeConfig = () => {
     dispatch(setFee({ inputName: inputName, inputValue: inputValue }))
   }
 
-  function handlerefferalCommission(
+  function handleRefferalCommission(
```
Comment (nicholascostadev) [inline src/templates/Manage/CreatePool/ConfigureFee/FeeConfig/index.tsx:50]: It is spelled "referral"

```suggestion
  function handleReferralCommission(
```

File: src/templates/PoolManager/FeeRewards/index.tsx [hunk 7]
```diff
@@ -135,17 +186,189 @@ const FeeRewards = () => {
...
+      feeData = {
+        ...feesData,
+        [inputName]: {
+          ...feesData[inputName],
+          isChecked: !feesData[inputName].isChecked
+        }
+      }
+    }
+
+    setFeesData(feeData)
```
Comment (nicholascostadev) [inline src/templates/PoolManager/FeeRewards/index.tsx:245]: This code is too complex and hard to follow, and you are multiple times checking the same thing(`if (inputName === x)`), so I would refactor it to use a switch case and save some checks.
Also, the `isChecked` can be a variable outside the ifs, here is the refactored code:

```suggestion
const inputName = event.target.name
  let isChecked = feesData[inputName].isChecked

  let feeData
  switch (inputName) {
    case "depositFee":
      if (isChecked) {
        feeData = {
          depositFee: {
            isChecked: false,
            feeRate: '0'
          },
          refferalFee: {
            isChecked: false,
            brokerCommision: 0,
            managerShare: 0
          }
        }
      } else {
        feeData = {
          depositFee: {
            isChecked: true,
            feeRate: currentDepositFee.toFixed()
          },
          refferalFee: {
            isChecked: true,
            brokerCommision: parseFloat(currentBrokerCommision.toFixed()),
            managerShare: parseFloat(currentManagerShare.toFixed())
          }
        }
      }
      break;
    case "referralFee":
      if (isChecked) {
        feeData = {
          ...feesData,
          refferalFee: {
            isChecked: false,
            brokerCommision: 0,
            managerShare: 0
          }
        }
      } else {
        feeData = {
          ...feesData,
          refferalFee: {
            isChecked: true,
            brokerCommision: parseFloat(currentBrokerCommision.toFixed()),
            managerShare: parseFloat(currentManagerShare.toFixed())
          }
        }
      }
      break;
    default:
      feeData = {
        ...feesData,
        [inputName]: {
          ...feesData[inputName],
          isChecked: !isChecked
        }
      }
  }

  setFeesData(feeData)
```

**Review by nicholascostadev — CHANGES_REQUESTED** [review]

**Review by guilherme-correa-s — APPROVED** [review]
LGTM
